### PR TITLE
Including more methods

### PR DESCRIPTION
**Pull request by filipecosta01**

### CODE REVIEW

File: app/controllers/telegram_webhooks_controller.rb [hunk 1]
```diff
@@ -97,4 +66,108 @@ def help(*)
   def action_missing(action)
     reply_with :message, text: "Can not perform #{action}" if command?
   end
-end
+
+  private
+  def getArgsFromCommand(* args)
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 Is there any specific reason for this space between the star symbol and the args?

File: app/controllers/telegram_webhooks_controller.rb [hunk 2]
```diff
@@ -9,74 +9,43 @@ def start(*)
     reply_with :message, text: 'Hi there!'
   end
 
-  def shows(*args)
-    endingDate = Date.parse(END_DATE_STRING, t(:date_format))
-
+  def shows(* args)
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 Is there any specific reason for this space between the star symbol and the args?

File: app/controllers/telegram_webhooks_controller.rb [hunk 3]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+
+    else
+      #Command does not match anything
+      errorMessage = MESSAGE_CHECK_USAGE_COMMAND
+
+    end
+
+    if not errorMessage
+      if isCurrentMonth
+        errorMessage = validateDateFromInput(dayFromArgs.to_i, getCurrentMonth(nil).to_i)
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: Sending nil as parameter seems a little bit odd to me (even though I got your idea). Wouldn't be better use Time.now.month?

File: app/controllers/telegram_webhooks_controller.rb [hunk 4]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+    if args[0].match(NUMBER_REGULAR_EXPRESSION)
+
+      dayFromArgs = args[0].to_s.rjust(2, '0')
+      isCurrentMonth = true
+
+    elsif args[0].match(DATE_REGULAR_EXPRESSION)
+
+      dateSplitted = args[0].split('/')
+      dateFormated = Date.strptime(dateSplitted[0] + "/" + dateSplitted[1] + "/" + getCurrentYear, t(:date_format))
+      puts dateFormated.month
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: Remove this

File: app/controllers/telegram_webhooks_controller.rb [hunk 5]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+    #Day informed as a number
+    if args[0].match(NUMBER_REGULAR_EXPRESSION)
+
+      dayFromArgs = args[0].to_s.rjust(2, '0')
+      isCurrentMonth = true
+
+    elsif args[0].match(DATE_REGULAR_EXPRESSION)
+
+      dateSplitted = args[0].split('/')
+      dateFormated = Date.strptime(dateSplitted[0] + "/" + dateSplitted[1] + "/" + getCurrentYear, t(:date_format))
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 I myself consider interpolation cleaner than concatenation. I'm not saying you have to change it here, but maybe you should take a look on how to do it so you can use it in the future.

Here we'd have something like: Date.strptime("#{dateSplitted[0]}/#{dateSplitted[1]}/#{getCurrentYear}", t(:date_format))

File: app/controllers/telegram_webhooks_controller.rb [hunk 6]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+      puts dateFormated.month
+      
+      dayFromArgs = dateFormated.strftime('%d')
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 Can't you extract the day using dateFormatted.day?

File: app/controllers/telegram_webhooks_controller.rb [hunk 7]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+      monthFromArgs = dateFormated.strftime('%m')
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 Can't you extract the month using dateFormatted.month?

File: app/controllers/telegram_webhooks_controller.rb [hunk 8]
```diff
@@ -10,73 +10,42 @@ def start(*)
...
-
-      dateFormated = t(:custom_date_format, :day => dayFromArgs, :month => currentMonth)
-
-      if @shows.empty?
-        reply_with :message, text: t(:empty_shows_results, :dateFormated => dateFormated)
       else
+        @shows = Show.by_date(requestedDate)
 
-        response_message = t(:shows_title_message, :dateFormated => dateFormated)
+        dateFormated = t(:custom_date_format, :day => requestedDate.strftime("%d"), :month => requestedDate.strftime("%m"))
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 dateFormatted*

File: app/controllers/telegram_webhooks_controller.rb [hunk 9]
```diff
@@ -10,73 +10,42 @@ def start(*)
...
-      if errorMessage
-        reply_with :message, text: errorMessage
+      if argsTypeErrorMessage
+        replyWithText(argsTypeErrorMessage)
         return
-      end
-
-      if not dayFromArgs
-        reply_with :message, text: t(:error_invalid_weekday)
+      elsif selectedDay.nil?
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: elsif not selectedDay

what do you think?

File: app/controllers/telegram_webhooks_controller.rb [hunk 10]
```diff
@@ -10,73 +10,42 @@ def start(*)
...
-      if Date.today > endingDate
-        reply_with :message, text: t(:ended_message)
+      if resultAvailable.is_a? String
+        replyWithText(resultAvailable)
         return
-      elsif requestedDate > endingDate
-        # If current date is bigger than the end date
-        reply_with :message, text: t(:error_after_end_date)
-        return
-      elsif errorMessage.nil?
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: elsif not errorMessage

what do you think?

File: app/controllers/telegram_webhooks_controller.rb [hunk 11]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+      return true
+    end
+  end
+
+  def getCurrentMonth(selectedDay)
+
+    currentMonth = Date.today.strftime("%m").to_s.rjust(2, '0')
+
+    # If the current day is bigger than the requested day, get data from next month
+    if selectedDay.nil?
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: if not selectedDay

what do you think?

File: app/controllers/telegram_webhooks_controller.rb [hunk 12]
```diff
@@ -97,4 +66,108 @@ def help(*)
...
+    elsif Date.today.strftime("%d").to_i > selectedDay.to_i
+      currentMonth = (Date.today.strftime("%m").to_i + 1).to_s.rjust(2, '0')
+    end
+
+    return currentMonth
+  end
+
+  def getRequestedDateFormated (selectedDay, selectedMonth)
+
+    if selectedMonth.nil?
```
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: if not selectedMonth

File: app/controllers/telegram_webhooks_controller.rb (hunk 5 above)
Comment (filipecosta01) [inline app/controllers/telegram_webhooks_controller.rb]: Nice tip, didn't know about it.

Fixed other stuff in a new push.

File: app/controllers/telegram_webhooks_controller.rb (hunk 9 above)
Comment (filipecosta01) [inline app/controllers/telegram_webhooks_controller.rb]: How bad is to use the current solution? Idk if it's just weird or something else...

**Comment by filipecosta01** [comment]
@felipevieira fixed code after your comments.

Please take a look again and thumbsup here if I can merge.

File: app/controllers/telegram_webhooks_controller.rb (hunk 9 above)
Comment (felipevieira) [inline app/controllers/telegram_webhooks_controller.rb]: @filipecosta01 maybe it's just me but I think that avoid using nil, null or other keywords helps on code 
readability.. but there's no obligation to change it right now

**Pull request merged**